src/main/java/org/jenkinsci/plugins/github/webhook/WebhookManager.java

```diff
@@ -141,8 +141,22 @@ public void run() {
      */
     public void unregisterFor(GitHubRepositoryName name, List<GitHubRepositoryName> aliveRepos) {
         try {
-            GHRepository repo = checkNotNull(
-                    from(name.resolve(allowedToManageHooks())).firstMatch(withAdminAccess()).orNull(),
```

Comment:
so null is a valid case here, but `checkNotNull` is throwing NPE. Just remove `checkNotNull` and return from method without anything.

Reply:
Optionally for debug purposes print into
```
   GHRepository repo = from(name.resolve(allowedToManageHooks())).firstMatch(withAdminAccess()).orNull();
   if (isNull(repo)) {
       LOGGER.debug("Skipped removing GitHub webhook for {} because not configured to Manage Hooks or "
                            + "there are no credentials with admin access to manage hooks", name);
       return;
   }

```

Reply:
I am not sure I got this comment properly, or my implementation change for that matter :)

I believe the original code had `null` as valid case of the not-finding-a-match via `orNull()`, and handling that via `checkNotNull()`.

My intention was to pick out the same match-or-not logic into the `repo` variable, which same way can end up `null`, but do the same `checkNotNull` handling after checking if we are at all configured to manage the hooks.

IIRC resolving of the `repo` with each and any call to register/unregister routines was also dictated by selftests somehow. My first shot was to check enablement of webhook management, log and return if it is not enabled - and then the tests failed until I changed the call chain to what is here.

Reply:
So registration of error should happen only for available allowedToManage repos but without admin access